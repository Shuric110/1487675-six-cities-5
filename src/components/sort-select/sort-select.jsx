import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {ActionCreator} from "../../store/action";
import {SortType, SORT_DEFINITIONS} from "../../const";
import withActiveFlag from "../../hocs/with-active-flag/with-active-flag";

const SortSelect = (props) => {
  const {sortType: currentSortType, setSortType, isActive, setIsActive} = props;

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" tabIndex="0" onClick={() => setIsActive(!isActive)}>
        {SORT_DEFINITIONS[currentSortType].title}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <ul className={`places__options places__options--custom ${isActive ? `places__options--opened` : ``}`}
        onClick={(evt) => {
          if (evt.target.dataset.sortType) {
            setSortType(evt.target.dataset.sortType);
            setIsActive(false);
          }
        }}
      >
        {Object.values(SortType).map((sortType) => (
          <li tabIndex="0" key={sortType}
            className={`places__option ${currentSortType === sortType ? `places__option--active` : ``}`}
            data-sort-type={sortType}
          >
            {SORT_DEFINITIONS[sortType].title}
          </li>
        ))}
      </ul>
    </form>
  );
};

SortSelect.propTypes = {
  sortType: PropTypes.string.isRequired,
  setSortType: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  setIsActive: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  sortType: state.sortType,
});

const mapDispatchToProps = (dispatch) => ({
  setSortType(sortType) {
    dispatch(ActionCreator.setSortType(sortType));
  },
});

export {SortSelect};
export default connect(mapStateToProps, mapDispatchToProps)(withActiveFlag(SortSelect));
