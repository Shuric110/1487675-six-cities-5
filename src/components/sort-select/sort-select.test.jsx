import React from "react";
import renderer from "react-test-renderer";

import {SortSelect} from "./sort-select";

const noop = () => {};

describe(`Render SortSelect`, () => {
  it(`SortSelect with sort type 1 in active state renders correctly`, () => {
    const tree = renderer
      .create(
          <SortSelect
            sortType="PRICE_DESC"
            setSortType={noop}
            isActive={true}
            setIsActive={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`SortSelect with sort type 2 in inactive state renders correctly`, () => {
    const tree = renderer
      .create(
          <SortSelect
            sortType="DEFAULT"
            setSortType={noop}
            isActive={false}
            setIsActive={noop}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
