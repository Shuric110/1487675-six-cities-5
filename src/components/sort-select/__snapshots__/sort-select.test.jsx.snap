// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render SortSelect SortSelect with sort type 1 in active state renders correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Price: high to low
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom places__options--opened"
    onClick={[Function]}
  >
    <li
      className="places__option "
      data-sort-type="DEFAULT"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort-type="PRICE_ASC"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option places__option--active"
      data-sort-type="PRICE_DESC"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort-type="RATING_DESC"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;

exports[`Render SortSelect SortSelect with sort type 2 in inactive state renders correctly 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
  <ul
    className="places__options places__options--custom "
    onClick={[Function]}
  >
    <li
      className="places__option places__option--active"
      data-sort-type="DEFAULT"
      tabIndex="0"
    >
      Popular
    </li>
    <li
      className="places__option "
      data-sort-type="PRICE_ASC"
      tabIndex="0"
    >
      Price: low to high
    </li>
    <li
      className="places__option "
      data-sort-type="PRICE_DESC"
      tabIndex="0"
    >
      Price: high to low
    </li>
    <li
      className="places__option "
      data-sort-type="RATING_DESC"
      tabIndex="0"
    >
      Top rated first
    </li>
  </ul>
</form>
`;
